Issue-435: Arregla warnings en tests

FedericoJurio opened the pull request:
Este PR inicializa correctamente los objetos que generan RuntimeWarnings por naive datetimes.

eduzen commented:
No me gusta lo de `git+https://github.com/nkunihiko/django-bootstrap3-datetimepicker/` pero por lo demás para mi está bien.

facundobatista commented on events/forms.py:
```diff
@@ -33,7 +33,7 @@ class EventForm(CrispyFormMixin):
         widget=DateTimePicker(
             options={
-                "format": "DD/MM/YYYY HH:ss:mm"
+                "format": "DD/MM/YYYY HH:mm"
```
¿Por qué no poner los segundos al final y ya?

FedericoJurio replied:
Puse mis dedos en movimiento antes que mi cerebro en funcionamiento

facundobatista commented on requirements.txt:
```diff
@@ -1,8 +1,9 @@
 django-allauth==0.31.0
 django-autoslug==1.9.3
-django-bootstrap3-datetimepicker==2.2.3
+django-braces==1.11.0
```
para qué es este?

FedericoJurio replied:
No recuerdo el motivo pero ya no está más

facundobatista commented on requirements.txt:
```diff
@@ -1,8 +1,9 @@
-django-bootstrap3-datetimepicker==2.2.3
+django-braces==1.11.0
 django-bootstrap3==8.2.1
+git+https://github.com/nkunihiko/django-bootstrap3-datetimepicker/
```
por qué usamos desde el proyecto y no la versión releaseada?

FedericoJurio replied:
Si mal no recuerdo el proyecto estaba deprecado pero el fix estaba ahi. En fin, no le demos muchas vueltas porque esto tampoco está más

facundobatista reviewed (changes requested):
Hola! Antes que nada perdón por el tiempo para hacer review de esto, se nos pasó totalmente.

Te hago un par de preguntas para entender mejor los cambios, gracias! Saludos.

FedericoJurio commented:
> Hola! Antes que nada perdón por el tiempo para hacer review de esto, se nos pasó totalmente.
> 
> Te hago un par de preguntas para entender mejor los cambios, gracias! Saludos.

@facundobatista Este PR es de 2018 y, si bien el proyecto no tiene mucho movimiento, algunas cosas cambiaron: actualmente en master hay menos warnings que en 2018. Por este motivo decidí hacer un reset a master del branch y solucionar los warnings que aparecieran

Ejecución de tests previo a mis cambios
```
make test
docker-compose run --rm web flake8
Creating pyarweb_web_run ... done
docker-compose run --rm web python3 ./manage.py test  -v2 --noinput
Creating pyarweb_web_run ... done
Creating test database for alias 'default' ('test_postgres')...
Operations to perform:
  Synchronize unmigrated apps: allauth, bootstrap3, bootstrap3_datetime, community, crispy_forms, dbbackup, disqus, django_extensions, email_obfuscator, messages, pagination, sanitizer, sendfile, staticfiles, taggit_autosuggest
  Apply all migrations: account, admin, auth, captcha, contenttypes, django_summernote, email_confirm_la, events, jobs, news, planet, pycompanies, sessions, sites, socialaccount, tagging, taggit
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying auth.0001_initial... OK
  Applying account.0001_initial... OK
  Applying account.0002_email_max_length... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying admin.0003_logentry_add_action_flag_choices... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_alter_group_name_max_length... OK
  Applying auth.0011_update_proxy_permissions... OK
  Applying captcha.0001_initial... OK
  Applying django_summernote.0001_initial... OK
  Applying django_summernote.0002_update-help_text... OK
  Applying email_confirm_la.0001_initial... OK
  Applying email_confirm_la.0002_auto_20170713_1447... OK
  Applying events.0001_initial... OK
  Applying events.0002_auto_20170325_2011... OK
  Applying events.0003_eventparticipation_gender... OK
  Applying events.0004_auto_20180428_1949... OK
  Applying events.0005_auto_20180429_1546... OK
  Applying taggit.0001_initial... OK
  Applying taggit.0002_auto_20150616_2121... OK
  Applying pycompanies.0001_initial... OK
  Applying jobs.0001_initial... OK
  Applying jobs.0002_auto_20170326_1616... OK
  Applying jobs.0003_normalize_tags_20180428_1955... Deleted 0 tags OK
  Applying news.0001_initial... OK
  Applying sites.0001_initial... OK
  Applying planet.0001_initial... OK
  Applying planet.0002_auto_20151219_2141... OK
  Applying pycompanies.0002_company_rank... OK
  Applying sessions.0001_initial... OK
  Applying sites.0002_alter_domain_unique... OK
  Applying socialaccount.0001_initial... OK
  Applying socialaccount.0002_token_max_lengths... OK
  Applying socialaccount.0003_extra_data_default_dict... OK
  Applying tagging.0001_initial... OK
  Applying tagging.0002_on_delete... OK
  Applying tagging.0003_adapt_max_tag_length... OK
  Applying taggit.0003_taggeditem_add_unique_index... OK
System check identified no issues (0 silenced).
test_a_recent_key_is_added_to_the_return_value (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_events_are_included_in_recent (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_included_events_have_correct_fields (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_included_jobs_have_correct_fields (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_included_news_have_correct_fields (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_items_are_sorted_by_the_created_field (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_jobs_are_included_in_recent (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_news_are_included_in_recent (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_recent_is_a_list_with_at_most_10_items (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_anonymous_user_confirmed_his_email_then_return_true (events.tests.test_event_participation.IsVerifiedTests) ... ok
test_anonymous_user_with_pending_email_confirmation_return_false (events.tests.test_event_participation.IsVerifiedTests) ... ok
test_user_is_not_none_then_return_true (events.tests.test_event_participation.IsVerifiedTests) ... ok
test_events_feed_view_list (events.tests.test_views.EventsViewTest) ... /usr/local/lib/python3.6/site-packages/django/db/models/fields/__init__.py:1368: RuntimeWarning: DateTimeField Event.start_at received a naive datetime (2021-11-09 22:16:21.364520) while time zone support is active.
  RuntimeWarning)
/usr/local/lib/python3.6/site-packages/django/db/models/fields/__init__.py:1368: RuntimeWarning: DateTimeField Event.end_at received a naive datetime (2021-11-10 22:16:21.364529) while time zone support is active.
  RuntimeWarning)
ok
test_events_view_create (events.tests.test_views.EventsViewTest) ... ok
test_events_view_delete (events.tests.test_views.EventsViewTest) ... ok
test_events_view_edit (events.tests.test_views.EventsViewTest) ... ok
test_events_view_list (events.tests.test_views.EventsViewTest) ... ok
test_events_view_slug (events.tests.test_views.EventsViewTest) ... ok
test_html_sanitizer_in_description_field (events.tests.test_views.EventsViewTest) ... ok
test_normalize_tags_with_non_ascii (jobs.tests.test_utils.UtilTest) ... ok
test_normalize_tags_with_repeated (jobs.tests.test_utils.UtilTest) ... ok
test_normalize_tags_with_repeated_and_spaces (jobs.tests.test_utils.UtilTest) ... ok
test_html_sanitizer_in_description_field (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_create (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_create_avoiding_repeated_tags (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_edit (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_idelete (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_list (jobs.tests.test_views.JobsTest) ... /usr/local/lib/python3.6/site-packages/django/db/models/fields/__init__.py:1368: RuntimeWarning: DateTimeField Job.created received a naive datetime (2021-09-09 22:16:23.845783) while time zone support is active.
  RuntimeWarning)
ok
test_jobs_view_list_regular_and_sponsored (jobs.tests.test_views.JobsTest) ... ok

----------------------------------------------------------------------
Ran 29 tests in 7.682s

OK
Destroying test database for alias 'default' ('test_postgres')...
```

Ejecución de tests posterior a mis cambios
```
make test
docker-compose run --rm web flake8
Creating pyarweb_web_run ... done
docker-compose run --rm web python3 ./manage.py test  -v2 --noinput
Creating pyarweb_web_run ... done
Creating test database for alias 'default' ('test_postgres')...
Operations to perform:
  Synchronize unmigrated apps: allauth, bootstrap3, bootstrap3_datetime, community, crispy_forms, dbbackup, disqus, django_extensions, email_obfuscator, messages, pagination, sanitizer, sendfile, staticfiles, taggit_autosuggest
  Apply all migrations: account, admin, auth, captcha, contenttypes, django_summernote, email_confirm_la, events, jobs, news, planet, pycompanies, sessions, sites, socialaccount, tagging, taggit
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying auth.0001_initial... OK
  Applying account.0001_initial... OK
  Applying account.0002_email_max_length... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying admin.0003_logentry_add_action_flag_choices... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_alter_group_name_max_length... OK
  Applying auth.0011_update_proxy_permissions... OK
  Applying captcha.0001_initial... OK
  Applying django_summernote.0001_initial... OK
  Applying django_summernote.0002_update-help_text... OK
  Applying email_confirm_la.0001_initial... OK
  Applying email_confirm_la.0002_auto_20170713_1447... OK
  Applying events.0001_initial... OK
  Applying events.0002_auto_20170325_2011... OK
  Applying events.0003_eventparticipation_gender... OK
  Applying events.0004_auto_20180428_1949... OK
  Applying events.0005_auto_20180429_1546... OK
  Applying taggit.0001_initial... OK
  Applying taggit.0002_auto_20150616_2121... OK
  Applying pycompanies.0001_initial... OK
  Applying jobs.0001_initial... OK
  Applying jobs.0002_auto_20170326_1616... OK
  Applying jobs.0003_normalize_tags_20180428_1955... Deleted 0 tags OK
  Applying news.0001_initial... OK
  Applying sites.0001_initial... OK
  Applying planet.0001_initial... OK
  Applying planet.0002_auto_20151219_2141... OK
  Applying pycompanies.0002_company_rank... OK
  Applying sessions.0001_initial... OK
  Applying sites.0002_alter_domain_unique... OK
  Applying socialaccount.0001_initial... OK
  Applying socialaccount.0002_token_max_lengths... OK
  Applying socialaccount.0003_extra_data_default_dict... OK
  Applying tagging.0001_initial... OK
  Applying tagging.0002_on_delete... OK
  Applying tagging.0003_adapt_max_tag_length... OK
  Applying taggit.0003_taggeditem_add_unique_index... OK
System check identified no issues (0 silenced).
test_a_recent_key_is_added_to_the_return_value (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_events_are_included_in_recent (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_included_events_have_correct_fields (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_included_jobs_have_correct_fields (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_included_news_have_correct_fields (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_items_are_sorted_by_the_created_field (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_jobs_are_included_in_recent (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_news_are_included_in_recent (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_recent_is_a_list_with_at_most_10_items (community.tests.test_homepage.GetContextDataRecentKeyTests) ... ok
test_anonymous_user_confirmed_his_email_then_return_true (events.tests.test_event_participation.IsVerifiedTests) ... ok
test_anonymous_user_with_pending_email_confirmation_return_false (events.tests.test_event_participation.IsVerifiedTests) ... ok
test_user_is_not_none_then_return_true (events.tests.test_event_participation.IsVerifiedTests) ... ok
test_events_feed_view_list (events.tests.test_views.EventsViewTest) ... ok
test_events_view_create (events.tests.test_views.EventsViewTest) ... ok
test_events_view_delete (events.tests.test_views.EventsViewTest) ... ok
test_events_view_edit (events.tests.test_views.EventsViewTest) ... ok
test_events_view_list (events.tests.test_views.EventsViewTest) ... ok
test_events_view_slug (events.tests.test_views.EventsViewTest) ... ok
test_html_sanitizer_in_description_field (events.tests.test_views.EventsViewTest) ... ok
test_normalize_tags_with_non_ascii (jobs.tests.test_utils.UtilTest) ... ok
test_normalize_tags_with_repeated (jobs.tests.test_utils.UtilTest) ... ok
test_normalize_tags_with_repeated_and_spaces (jobs.tests.test_utils.UtilTest) ... ok
test_html_sanitizer_in_description_field (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_create (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_create_avoiding_repeated_tags (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_edit (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_idelete (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_list (jobs.tests.test_views.JobsTest) ... ok
test_jobs_view_list_regular_and_sponsored (jobs.tests.test_views.JobsTest) ... ok

----------------------------------------------------------------------
Ran 29 tests in 7.925s

OK
Destroying test database for alias 'default' ('test_postgres')...
```

facundobatista commented:
Hola! Te pido un último empujón con este PR... al final terminamos sacando "datetimepicker" porque no encontramos una combinación de versiones que no esté roto.

Entonces, creo que solucionás ese conflicto y esto está para landear... 

Muchas graicas!!!

FedericoJurio commented:
@facundobatista Conflicto resuelto

facundobatista reviewed (approved):
Nice! Thanks
